### ISSUE TITLE
Selection in VI visual mode need to include the last character (+1 grapheme to the right)

### ISSUE THREAD

**Issue by blindFS** [nushell/reedline#843]
**Platform** macOS
**Terminal software** wezterm

https://github.com/user-attachments/assets/57347516-eeed-441e-b1b8-02cb1f1111de

This is not what a traditional vim behavior where current character "a" should be included in both directions.
Basically, the end offset returned by get_selection should be increased to the next grapheme.

## Steps to reproduce
1. set editor mode to vi
2. press v to visual selection

Seems that crate::core_editor::Editor should be aware of current edit mode in order to fix this issue.

**Comment by fdncred** [comment]
seems like a reasonable request. i also wish the coloring was maybe just the reverse of whatever the text is. it's so hard to see, but that's a separate issue.

**Comment by blindFS** [comment]
> seems like a reasonable request. i also wish the coloring was maybe just the reverse of whatever the text is. it's so hard to see, but that's a separate issue.

Yes, there's exposed API `with_visual_selection_style` in engine.rs but never get called in nushell.

**Comment by fdncred** [comment]
https://github.com/nushell/nushell/pull/14052

**Comment by sholderbach** [comment]
This should be closed by #867
